test/framework/envs/multizone/env.go

```diff
@@ -218,3 +231,14 @@ func RestoreState(bytes []byte) {
 	Expect(UniZone2.AddNetworking(state.UniZone2.ZoneEgress, Config.ZoneEgressApp)).To(Succeed())
 	Expect(UniZone2.AddNetworking(state.UniZone2.ZoneIngress, Config.ZoneIngressApp)).To(Succeed())
 }
+
+func getKumaDeploymentOptions(config framework.ControlPlaneConfig, defaultOptions ...framework.KumaDeploymentOption) []framework.KumaDeploymentOption {
+	kumaOptions := append([]framework.KumaDeploymentOption{}, defaultOptions...)
+	for key, value := range config.Envs {
+		kumaOptions = append(kumaOptions, framework.WithEnv(key, value))
+	}
+	if config.AdditionalYamlConfig != "" {
+		kumaOptions = append(kumaOptions, framework.WithYamlConfig(config.AdditionalYamlConfig))
+	}
+	return kumaOptions
```

Comment:
nit: you could create a function in framework package
```
func KumaDeploymentOptionsFromConfig(config framework.ControlPlaneConfig) []framework.KumaDeploymentOption {
	kumaOptions := []framework.KumaDeploymentOption{}
	for key, value := range config.Envs {
		kumaOptions = append(kumaOptions, framework.WithEnv(key, value))
	}
	if config.AdditionalYamlConfig != "" {
		kumaOptions = append(kumaOptions, framework.WithYamlConfig(config.AdditionalYamlConfig))
	}
	return kumaOptions
}
```
and use it in all deployments variants.